pkg/collector/corechecks/snmp/report/report_device_metadata.go

```diff
@@ -73,11 +73,21 @@ func buildMetadataStore(metadataConfigs checkconfig.MetadataConfig, values *valu
 				}
 			} else {
 				for _, symbol := range symbols {
-					metricValues, err := getColumnValueFromSymbol(values, symbol)
+					indexValues := getColumnValueFromSymbolOrEmpty(symbol.IndexFromOidValue.OID, values)
+					fieldValues, err := getColumnValueFromSymbol(values, symbol)
 					if err != nil {
 						continue
 					}
-					for fullIndex, value := range metricValues {
+					for fullIndex, value := range fieldValues {
+						newIndexVal, ok := indexValues[fullIndex]
```

Comment:
Sounds great if there is a 1-1 matching between the symbol.OID and symbol.IndexFromOidValue.OID, but in case of ip addresses for instance, there is a one-to-many relationship. We probably need to support something like this